src/azure-cli/azure/cli/command_modules/containerapp/_up_utils.py

```diff
@@ -597,7 +597,16 @@ def run_acr_build(self, dockerfile, source, quiet=False, build_from_source=False
         now = datetime.now()
         tag_now_suffix = str(now).replace(" ", "").replace("-", "").replace(".", "").replace(":", "")
         image_name_with_tag = image_name + ":{}".format(tag_now_suffix)
-        self.image = self.registry_server + "/" + image_name_with_tag
+
+        if ":" in image_name.split("/")[-1]:
+            image_name_with_tag = image_name
+        else:
+            image_name_with_tag = image_name + ":{}".format(tag_now_suffix)
+
+        if not image_name_with_tag.startswith(self.registry_server):
```

Comment:
any code to validate if the image_name_with_tag's server match the registry_server?

Reply:
![image](https://github.com/user-attachments/assets/78f4c8d4-1be7-4597-ac1a-672d4cc3c73d)

Just add validation for this check.

Reply:
Sync offline, not need to check.

Accepct:
2024/09/19 07:27:33 Successfully pushed image: xinyutest2.azurecr.io/caedd971c2c1acr.azurecr.io/k8se/quickstart:latest